openwisp_notifications/tests/test_api.py

```diff
@@ -592,8 +595,40 @@ def test_list_notification_setting_filtering(self):
             ns = response.data['results'].pop()
             self.assertEqual(ns['type'], 'default')
 
+        with self.subTest('Test without authenticated'):
+            self.client.logout()
+            user_url = self._get_path('user_notification_setting_list', tester.pk)
+            response = self.client.get(user_url)
+            self.assertEqual(response.status_code, 401)
+
+        with self.subTest('Test filtering by user_id as admin'):
+            self.client.force_login(self.admin)
+            user_url = self._get_path('user_notification_setting_list', tester.pk)
+            response = self.client.get(user_url)
+            self.assertEqual(response.status_code, 200)
+
+        with self.subTest('Test with user_id by user_id as the same user'):
+            self.client.force_login(tester)
+            user_url = self._get_path('user_notification_setting_list', tester.pk)
+            response = self.client.get(user_url)
+            self.assertEqual(response.status_code, 200)
+
+        with self.subTest('Test with user_id as a different non-admin user'):
+            self.client.force_login(tester)
+            user_url = self._get_path('user_notification_setting_list', self.admin.pk)
+            response = self.client.get(user_url)
+            self.assertEqual(response.status_code, 403)
+
     def test_retreive_notification_setting_api(self):
-        notification_setting = NotificationSetting.objects.first()
+        notification_setting = NotificationSetting.objects.exclude(
+            organization__isnull=True
+        ).first()
+        tester = self._create_user()
+        tester_notification_setting = NotificationSetting.objects.create(
+            user=tester,
+            type='default',
+            organization=Organization.objects.first(),
+        )
```

Comment:
You wouldn't need to create the notification setting manually if you create an organization admin here instead.